### PR TITLE
{WIP} make string and buffer usages safer

### PR DESCRIPTION
**Pull request by mtuncer**
We are replacing unsafe usages of string functions and memory operations with safer ones.

### CODE REVIEW

File: src/bin/pg_autoctl/cli_common.c [hunk 1]
```diff
@@ -197,8 +197,8 @@ cli_create_node_getopts(int argc, char **argv,
 			case 'g':
 			{
 				/* { "group", required_argument, NULL, 'g' } */
-				int scanResult = sscanf(optarg, "%d", &LocalOptionConfig.groupId);
-				if (scanResult == 0)
+				LocalOptionConfig.groupId = strtol(optarg, NULL, 10);
+				if (errno != 0)
```
Comment (JelteF) [inline src/bin/pg_autoctl/cli_common.c]: afaict this error will not be hit in the following cases:
```
--group ''
--group '123somethingelse'
```

This is why I used these function in citus: https://github.com/citusdata/citus/pull/3602

I think it makes sense to have a similar function in this codebase, so small mistakes like this don't happen:
```c
bool StringToInt(const char *str, int *result)
```

Comment (mtuncer) [reply src/bin/pg_autoctl/cli_common.c]: I did not change the sscanf behavior there. We seem to print different error messages depending on the context

agree on our own StringToInt function.

File: src/bin/pg_autoctl/main.c [hunk 2]
```diff
@@ -13,6 +13,7 @@
 #include "postgres_fe.h"
 
 #include "cli_root.h"
+#include "env_utils.h"
```
Comment (JelteF) [inline src/bin/pg_autoctl/main.c]: Seems this is not included in this PR

Comment (mtuncer) [reply src/bin/pg_autoctl/main.c]: added

**Comment by JelteF** [comment]
Replaced by a lot of separate PRs

**Pull request closed without merging**